Sources/BrowserServicesKit/Autofill/AutofillUserScript+SecureVault.swift

```diff
@@ -433,19 +447,45 @@ extension AutofillUserScript {
 
     func getAvailableInputTypes(_ message: UserScriptMessage, _ replyHandler: @escaping MessageReplyHandler) {
         let domain = hostForMessage(message)
+        Self.domainOfMostRecentGetAvailableInputsMessage = domain
         let email = emailDelegate?.autofillUserScriptDidRequestSignedInStatus(self) ?? false
-        vaultDelegate?.autofillUserScript(self, didRequestAutoFillInitDataForDomain: domain) { credentials, identities, cards, credentialsProvider  in
+        vaultDelegate?.autofillUserScript(self, didRequestAutoFillInitDataForDomain: domain) { [weak self] credentials, identities, cards, credentialsProvider, totalCredentialsCount in
+            guard let self else {
+                replyHandler("")
+                return
+            }
+            let credentialsImport = self.shouldShowPasswordImportDialog(credentials: credentials, credentialsProvider: credentialsProvider, totalCredentialsCount: totalCredentialsCount)
             let response = RequestAvailableInputTypesResponse(credentials: credentials,
                                                               identities: identities,
                                                               cards: cards,
                                                               email: email,
-                                                              credentialsProvider: credentialsProvider)
+                                                              credentialsProvider: credentialsProvider,
+                                                              credentialsImport: credentialsImport)
             if let json = try? JSONEncoder().encode(response), let jsonString = String(data: json, encoding: .utf8) {
                 replyHandler(jsonString)
             }
         }
     }
 
+    private func shouldShowPasswordImportDialog(credentials: [SecureVaultModels.WebsiteCredentials], credentialsProvider: SecureVaultModels.CredentialsProvider, totalCredentialsCount: Int) -> Bool {
+        guard credentialsProvider.name != .bitwarden else {
```

Comment:
One more condition for you! When testing with “Ask to Save and Autofill” disabled for Passwords in Setttings, I get the prompt to import passwords, which isn’t a great experience as after importing you never get prompted to autofill with your newly imported passwords